### ISSUE TITLE
Remove OIF substepping from ExaDG

### ISSUE THREAD

**Issue by nfehn** [exadg/exadg#158]
I do not see a benefit in OIF (operator-integration-factor) substepping and, therefore, plan to remove it from ExaDG.

**Comment by kronbichler** [comment]
I agree on this topic, given where we plan to put the resources in the coming years. We still have the code around from the current version (hopefully in a way that does not require a complete re-implementation even after the main refactoring we're going through right now) in case we feel the functionality should be re-established or someone wants to spend time on it.
